joomla/joomla-cms
# [4.0] Use new filter.condition instead of filter.published for article modules

### Summary of Changes
While testing #24672 I noticed that unpublished articles were visible. After some research, it seems to be caused by the new Joomla Workflow implementation. 

In the `ArticlesModel` the filter `filter.published` is replaced by `filter.condition`. So any code that is using  `filter.published` as a filter to load articles won't work and needs to be replaced by   `filter.condition`.

This PR fixes this for `mod_articles_category`, `mod_articles_news` and `mod_articles_populair`.

### Testing Instructions
Make sure to have some unpublished articles in a category. Add modules at a position in your template for `mod_articles_category`, `mod_articles_news` and `mod_articles_populair` and configure them in a way that the unpublished articles are visible. 

Apply the patch, and the unpublished articles should no longer be displayed (if you are not logged in on the frontend of the site).

### Expected result
Unpublished articles should not be visible.

### Actual result
Unpublished articles are visible.
